### PR TITLE
[8.0][FIX] Email templates for event registrations

### PR DESCRIPTION
**Pull request by JavierIniesta**
'event.registration' added to model_id domain in email template form view

### CODE REVIEW

File: event_registration_mass_mailing/views/email_template.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<openerp>
+<data>
+
+<record id="email_template_form_minimal" model="ir.ui.view">
+    <field name="name">Email Template - model_id domain</field>
+    <field name="model">email.template</field>
+    <field name="inherit_id" ref="mass_mailing.email_template_form_minimal"/>
+    <field name="arch" type="xml">
+        <field name="model_id" position="attributes">
+            <attribute name="domain">[('model', 'in', ['res.partner', 'mail.mass_mailing.contact', 'event.registration'])]</attribute>
```
Comment (pedrobaeza) [inline event_registration_mass_mailing/views/email_template.xml]: This makes that another module making the same overwrites your domain. You should inject it in the fields_view_get method instead.

**Comment by JavierIniesta** [comment]
Done. Method fields_view_get inherited.

**Comment by rafaelbn** [comment]
Tested, this is a FIX of event_registration_mass_mailing and must be merged as soon as possible
:+1:

**Comment by rafaelbn** [comment]
Hi all,

Please we need this quickly

Could and of you review?

Thanks a lot

Regards,
Rafael

File: event_registration_mass_mailing/models/email_template.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            toolbar=toolbar, submenu=submenu)
+        if 'arch' in res:
+            doc = etree.XML(res['arch'])
+            for node in doc.xpath("//field[@name='model_id']"):
+                domain = node.get('domain')
+                if domain:
+                    domain = eval(domain)
+                    for n, crit in enumerate(domain):
+                        if crit[0] == 'model' and crit[1] == 'in':
+                            crit[2].append('event.registration')
```
Comment (antespi) [inline event_registration_mass_mailing/models/email_template.py]: Here you could have a tuple instead a list. In that case, this error will raise:

```
AttributeError: 'tuple' object has no attribute 'append'
```

So I suggest this code:

```
for n, crit in enumerate(domain):
   if crit[0] == 'model' and crit[1] == 'in':
       v = list(crit[2])
       v.append('event.registration')
       domain[n] = ('model', 'in', tuple(v))
```

**Comment by JavierIniesta** [comment]
@antespi's suggestion applied and test added

File: event_registration_mass_mailing/models/email_template.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                        toolbar=False, submenu=False):
+        res = super(EmailTemplate, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        if 'arch' in res:
+            doc = etree.XML(res['arch'])
+            for node in doc.xpath("//field[@name='model_id']"):
+                domain = node.get('domain')
+                if domain:
+                    domain = eval(domain)
```
Comment (antespi) [inline event_registration_mass_mailing/models/email_template.py]: Use a safe_eval method:

```
from openerp.tools.safe_eval import safe_eval
(...)
if domain:
    domain = safe_eval(domain)
(...)
```

File: event_registration_mass_mailing/tests/test_email_template.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+from openerp.tests.common import TransactionCase
+
+
+class TestEmailTemplate(TransactionCase):
+
+    def test_fields_view_get(self):
+        self.env['email.template'].with_context(
+            {'form_view_ref': 'mass_mailing.email_template_form_minimal'}
+        ).fields_view_get()
```
Comment (antespi) [inline event_registration_mass_mailing/tests/test_email_template.py:16]: Add an assert to check that `fields_view_get` is adding element in domain

File: event_registration_mass_mailing/models/email_template.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                    for n, crit in enumerate(domain):
+                        if crit[0] == 'model' and crit[1] == 'in':
+                            v = list(crit[2])
+                            v.append('event.registration')
```
Comment (antespi) [inline event_registration_mass_mailing/models/email_template.py]: Check that 'event.registration' is not already in list:

```
v = list(crit[2])
if 'event.registration' not in v:
    v.append('event.registration')
```

**Comment by JavierIniesta** [comment]
Updated

**Comment by antespi** [comment]
:+1: Cool, thanks
code reviewed and tested in runbot

**Comment by janetrp** [comment]
tested in runbot :+1:

**Pull request merged**